src/mlagility/api/execute_trt.py

```diff
@@ -115,6 +126,65 @@ def format_field(line: str):
         e.writelines(decoded_error)
 
 
+"""
+In the average_power_and_utilization function, we eliminate the values below
+the idle_threshold at the beginning and end of the list because we want to 
+exclude the idle periods before and after the actual workload execution. By
+doing this, we can get a more accurate representation of the average power 
+consumption and GPU utilization during the execution of the workload.
+We set the Idle threshold to 3% utilization based on heuristics.
+"""
+IDLE_THRESHOLD = 3
+
+
+def average_power_and_utilization(power_readings):
+
+    if not power_readings:
+        return 0, 0
+
+    # Remove readings below the threshold from the beginning of the list
+    while power_readings and power_readings[0][0] <= IDLE_THRESHOLD:
+        power_readings.pop(0)
+
+    # Remove readings below the threshold from the end of the list
+    while power_readings and power_readings[-1][0] <= IDLE_THRESHOLD:
+        power_readings.pop()
+
+    if not power_readings:
+        return 0, 0
+
+    total_power = sum(power_draw for _, power_draw in power_readings)
```

Comment:
nit: the variable name `total_power` makes me think this is the power across multiple components or something. The `sum(power_draw)` over some samples is closer to the total energy used in the operation (but not quite, since it's sample-based). 

Suggest renaming this variable to `sum_of_power_samples` or something similarly precise.

Reply:
Or just eliminate the variable name altogether and do:

```python
average_power = sum(power_draw for _, power_draw in power_readings) / len(power_readings)
```

Reply:
Done, eliminated `total_power`.

Reply:
@ramkrishna2910 did you change push to server? I still see `total_power`